onnxruntime/core/providers/cuda/cu_inc/common.cuh

```diff
@@ -291,5 +291,8 @@ struct GridDim {
   if (id >= N)                                     \
     return;
 
+#define ORT_MAX_32F (3.402823466e+38f)
```

Comment:
https://stackoverflow.com/questions/8812422/how-to-find-epsilon-min-and-max-constants-for-cuda
Single max value (CUDA - approx):
CUDA = 1.70141e+38, CPU = 3.40282e+38

Reply:
std::numeric_limits<float>::infinity() can be used in CUDA kernel.

Reply:
std::numeric_limits::infinity is a host function. It can not be called from a device function.

Reply:
> https://stackoverflow.com/questions/8812422/how-to-find-epsilon-min-and-max-constants-for-cuda
> Single max value (CUDA - approx):
> CUDA = 1.70141e+38, CPU = 3.40282e+38
> How about using CUDART_MAX_NORMAL_F defined in CUDA?
The calculation in the stachoverflow is not accurate by multiple 2 each time until overflow. It should be ~3.40282e+38, 1.70141e+38 is too small.
Use CUDART_MAX_NORMAL_F as CUDA already defines it

Reply:
> std::numeric_limits::infinity is a host function. It can not be called from a device function.

It can. You just need to add compiling option "--expt-relaxed-constexpr".

Reply:
> expt-relaxed-constexpr



> std::numeric_limits::infinity is a host function. It can not be called from a device function.
> 
> It can. You just need to add compiling option "--expt-relaxed-constexpr".

Good to know that. As cuda has its own defined max, won't introduce it in this PR.